Backend/src/ml_lib/shared/shared_model.py

```diff
@@ -22,7 +22,7 @@ def _format_data(self, file_name: str) -> None:
         """
         pass
 
-    def run(self, file_name: str, is_first: bool) -> None:
+    def run(self, file_name: str, is_first: bool, dim: str = None) -> None:

```

Comment:
I think that the type for "dim" should be `str | None` . This also applies in the sub_models